libs/feature/search/src/lib/favorites/favorite-star/favorite-star.component.ts

```diff
@@ -33,17 +47,55 @@ export class FavoriteStarComponent {
   record_: MetadataRecord
   favoriteCount: number | null
   loading = false
+  DEFAULT_GN4_LOGIN_URL = `/geonetwork/srv/${
+    LANG_2_TO_3_MAPPER[this.translateService.currentLang]
+  }/catalog.signin?redirect=`
+  loginUrl = `${
+    this.optionalLoginUrl ? this.optionalLoginUrl : this.DEFAULT_GN4_LOGIN_URL
+  }${window.location}`
+  loginMessage = this.translateService.instant(
+    'favorite.not.authenticated.tooltip',
+    {
+      link: this.loginUrl,
+    }
+  )
+  @ViewChild(StarToggleComponent, { read: ElementRef })
+  starToggleRef: ElementRef
+  subscription: Subscription
 
   get hasFavoriteCount() {
     return this.favoriteCount !== null
   }
 
   constructor(
+    @Optional()
```

Comment:
Once the token is given a default value the `@Optional` decorator is not needed any more

Reply:
Couldn't get the default value to be provided by the token, so I kept it `@Optional`